htseq2multx.pl

```diff
@@ -0,0 +1,884 @@
+#!/usr/bin/env perl
+
+use warnings;
+use strict;
+use Getopt::Long qw(:config no_auto_abbrev);
+use IO::Pipe::Producer;
+use IO::Select;
+
+our $VERSION     = '0.1';
+our $BCSVERSION  = '0.18.6';
+our $FQMXVERSION = '1.4';
+
+my $help            = 0;
+my $version         = 0;
+my $idxread         = [];
+my $mismatches      = 1;
+my $barcodes_at_end = 0;
+my $galaxy          = 0;
+my $sanitize        = 1;
+my $verbose         = 0;
+my $gzipout         = 0;
+my $split_all       = 0;
+my $fastq_files     = [];
+my $index_hash      = {};
+my $bchash          = {};
+my $samplehash      = {};
+my $fastq_multx     = 'fastq-multx';
+my $debug           = 0;
+my $check_deflines  = 0;
+my $tmp_files       = [];
+my($bcfile,$bcfilefqmx,$stodout,$prefix,$suffix,$format,$gzipin,$deflinesep);
+
+main();
+
+sub main
+  {
+    processOptions();
+
+    my $command = getFastqMultxCommand();
+
+    if($debug)
+      {print STDERR ("#$command\n")}
+
+    runFastqMultx($command);
+  }
+
+
+
+
+
+
+##
+## Methods
+##
+
+sub processOptions
+  {
+    ##
+    ## Parse the command line and handle help/version options
+    ##
+
+    my $OptionHash = {
+                      'version:+'       => \$version,
+                      'h|help:+'        => \$help,
+                      'bcfile=s'        => \$bcfile,
+                      'idxread=i{1,2}'  => $idxread,
+                      'mismatches=i'    => \$mismatches,
+                      'barcodes_at_end' => \$barcodes_at_end,
+                      'prefix=s'        => \$prefix,
+                      'suffix=s'        => \$suffix,
+                      'galaxy'          => \$galaxy,
+                      'sanitize!'       => \$sanitize,
+                      'v|verbose:+'     => \$verbose,
+                      'gzipout'         => \$gzipout,
+                      'split_all'       => \$split_all,
+                      'format=s'        => \$format,
+                      'gzipin'          => \$gzipin,
+                      '<>'              => sub {push(@$fastq_files,$_[0])},
```

Comment:
Minor, but my general preference is for (for clarity of deref precedence):

```suggestion
                      '<>'              => sub {push(@{ $fastq_files },$_[0])},
```